### PR TITLE
Adding a new book which now comes with a revised Edition (java)

### PR DESCRIPTION
**Pull request by hrishikeshp7**
Introduction to Programming by Java by David J. Eck . 
Please Note that This one is Revised and Corrected Edition from the Well Known University of IIT Kanpur

### CODE REVIEW

File: books/free-programming-books.md [hunk 1]
```diff
@@ -1692,6 +1692,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Introduction to Computer Science "booksite"](https://introcs.cs.princeton.edu/java/cs/) - by Robert Sedgewick and Kevin Wayne (HTML)
 * [Introduction to Computer science using Java](http://www.programmedlessons.org/Java9/index.html) - Bradley Kjell
 * [Introduction to Programming in Java](http://introcs.cs.princeton.edu/java/home/) - Robert Sedgewick and Kevin Wayne
+* [Introduction to Programming Using Java] (https://www.iitk.ac.in/esc101/share/downloads/javanotes5.pdf) - David Eck (Revised Version)
```
Comment (davorpa) [inline books/free-programming-books.md:1695]:
```suggestion
* [Introduction to Programming Using Java (5th Edition - revised Nov 2007)] (https://www.iitk.ac.in/esc101/share/downloads/javanotes5.pdf) - David J. Eck (PDF)
```
👮 Following guidelines:
- Include edition/older publish date of resources as part of the title.
- Provide support format.
- Maybe search book in official website: http://math.hws.edu/javanotes (5th edition at https://math.hws.edu/eck/cs124/javanotes5/)

😉

**Review by davorpa — CHANGES_REQUESTED** [review]
The web site for this book is: http://math.hws.edu/javanotes

Currently at 8th Edition
All editions contains extra resources like exercises provided as zip.

5th Edition: https://math.hws.edu/eck/cs124/javanotes5/

Remember change status to Ready to review when draft is done 🤗

File: books/free-programming-books.md (hunk 1 above)
Comment (hrishikeshp7) [reply books/free-programming-books.md]: Hey @davorpa , Thanks for your help but I dont know why the Travis Build Isnt Passing , There is somewhere an Alphabetical Order mistake so I am bit of confused

**Comment by eshellman** [comment]
If we have the 8th edition, why do we need a version of the 5th edition?
Be sure to check the output of Travis-CI for linter errors. Alphabetical order and spacing are the most common problems. there are no annotations that I can see in the new link explaining any changes.

File: books/free-programming-books.md (hunk 1 above)
Comment (davorpa) [reply books/free-programming-books.md]: Dont be afraid. It can be changed using suggestions

File: books/free-programming-books.md [hunk 2]
```diff
@@ -1692,6 +1692,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Introduction to Computer Science "booksite"](https://introcs.cs.princeton.edu/java/cs/) - by Robert Sedgewick and Kevin Wayne (HTML)
 * [Introduction to Computer science using Java](http://www.programmedlessons.org/Java9/index.html) - Bradley Kjell
 * [Introduction to Programming in Java](http://introcs.cs.princeton.edu/java/home/) - Robert Sedgewick and Kevin Wayne
+* [Introduction to Programming Using Java (5th Edition - revised Nov 2007)] (https://www.iitk.ac.in/esc101/share/downloads/javanotes5.pdf) - David J. Eck (PDF)
 * [Introduction to Programming Using Java](http://math.hws.edu/javanotes/) - David J. Eck
```
Comment (davorpa) [inline books/free-programming-books.md:1696]:
```suggestion
* [Introduction to Programming Using Java](http://math.hws.edu/javanotes) - David J. Eck (HTML, PDF, ePUB + exercises)
* [Introduction to Programming Using Java (5th Edition - final version, 2010 Jun)](https://math.hws.edu/eck/cs124/javanotes5) - David J. Eck (HTML, PDF, ePUB + exercises)
```

- Solves CI/CD linting errors
- Reference to author site and lastest 5th edition

Comment (davorpa) [reply books/free-programming-books.md:1696]: @hrishikeshp7 take in mind resolve this suggested changes and make PR ready to review if you want advance to be merged 😉

**Review by davorpa — CHANGES_REQUESTED** [review]
Pending to resolve code change suggestions

**Review by davorpa — APPROVED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books, @hrishikeshp7 !

**Pull request merged**